### ISSUE TITLE
ObjectID string interface

### ISSUE THREAD

**Issue by hacksparrow** [strongloop/loopback-connector-mongodb#598]
All model properties defined as `{type: String, mongodb: {dataType: 'objectID'}` should be stored as `ObjectID` in the database but provide a string interface in the connector APIs. Meaning, string version of the `ObjectID` will be used in queries and results, `ObjectID` will not be directly exposed to users.

Any property not defined as `{type: String, mongodb: {dataType: 'objectID'}` will no be stored as `ObjectID`, even if the string looks like an ObjectID.

## Acceptance criteria

- [ ] Add implementation for `String` to `ObjectID` conversion in queries
- [ ] Add implementation for `ObjectID` to `String` conversion in results
- [ ] Refactor existing implementation wherever applicable
- [ ] Remove redundant code
- [ ] Update docs
- [ ] Add tests
- [ ] Add a "Long Term Support" section to README (see [strongloop/loopback](https://github.com/strongloop/loopback#module-long-term-support-policy) for inspiration) and also explain which connector versions are compatible with which LoopBack versions.

**Comment by bajtos** [comment]
Important: we already have a similar implementation in place for `Decimal128` type. Going forward, we should have a single unified solution for dealing with string-based data-types (`ObjectID`, `Decimal128`), to ensure consistent handling of edge cases like nested properties in anonymous models, nested array items, etc.

**Comment by hacksparrow** [comment]
Yes, in the proposed approach `ObjectID` and `Decimal128` coercion happen in the same block. It can be further optimized.

**Comment by mikeevstropov** [comment]
@hacksparrow When you said
`Any property not defined as {type: String, mongodb: {dataType: 'objectID'} will no be stored as ObjectID, even if the string looks like an ObjectID.`
Does CLI will set `dataType` for relation automatically? It looks like i need to handle it manually each time when i generate the relation. :hushed:

**Comment by paviln** [comment]
Hello,

When I use find on a repo, which model has a belongsTo:

  // Foreign key
  @belongsTo(
    () => User,
    {}, //relation metadata goes in here
    {
      // property definition goes in here
      type: 'string',
      required: true,
      mongodb: {dataType: 'ObjectId'},
    },
  )
  userId: string;

The result userId type is a object and not a string. This is annoying, since I want to use where neq..

The userId is stroed as ObjectId, if i change it to a string, I get a string value. But it should provide it as a string, even if it is a objectId?
